### PR TITLE
ci: enable pipeline in Jenkins with a non dockerised environment

### PR DESCRIPTION
**Pull request by v1v**
## Highlights
- Current test cases don't work when running in a docker container.
- Let's move to use nvm rather than docker and docker-compose.
- Use `nvm` to be able to switch node versions.
- Add build badge to the README.md.
- Fix some minor leftovers from the initial commit.

```
not ok 1196 should be equivalent
Stacktrace
      ---
      operator: deepEqual
    expected: |-
      { metadata: { service: { name: 'custom-serviceName', environment: 'production', version: 'custom-serviceVersion', runtime: { name: 'node', version: '12.6.0' }, language: { name: 'javascript' }, agent: { name: 'custom-agentName', version: 'custom-agentVersion' }, framework: { name: 'custom-frameworkName', version: 'custom-frameworkVersion' } }, process: { pid: 85, ppid: 78, title: '/usr/local/bin/node', argv: [ '/usr/local/bin/node', '/app/node_modules/.bin/tape', 'test/abort.js', 'test/basic.js', 'test/config.js', 'test/edge-cases.js', 'test/k8s.js', 'test/side-effects.js', 'test/stringify.js', 'test/truncate.js', 'test/writev.js' ] }, system: { hostname: 'custom-hostname', architecture: 'x64', platform: 'linux' }, labels: { foo: 'bar', doesNotNest: '[object Object]' } } }
    actual: |-
      { metadata: { service: { name: 'custom-serviceName', environment: 'production', runtime: { name: 'node', version: '12.6.0' }, language: { name: 'javascript' }, agent: { name: 'custom-agentName', version: 'custom-agentVersion' }, framework: { name: 'custom-frameworkName', version: 'custom-frameworkVersion' }, version: 'custom-serviceVersion' }, process: { pid: 85, ppid: 78, title: '/usr/local/bin/node', argv: [ '/usr/local/bin/node', '/app/node_modules/.bin/tape', 'test/abort.js', 'test/basic.js', 'test/config.js', 'test/edge-cases.js', 'test/k8s.js', 'test/side-effects.js', 'test/stringify.js', 'test/truncate.js', 'test/writev.js' ] }, system: { hostname: 'custom-hostname', architecture: 'x64', platform: 'linux', container: { id: '9a1066b98928fc2c2db476325991662054110182c8fca713892f860e585668ba' } }, labels: { foo: 'bar', doesNotNest: '[object Object]' } } }
    at: DestroyableTransform.<anonymous> (/app/test/config.js:173:9)
```

### CODE REVIEW

File: .ci/scripts/convert_tap_to_junit.sh [hunk 1]
```diff
@@ -9,18 +9,20 @@ npm install -g tap-junit
...
-for tf in $(ls ${TARGET_FOLDER}/*-output.tap)
+for tf in "${TARGET_FOLDER}"/*-output.tap
 do
-  filename=$(basename ${tf} ${TARGET_FOLDER}/output.tap)
-  if [ -s ${tf} ]; then
-    cat ${tf}|tap-junit --package="Agent Node.js" > ${TARGET_FOLDER}/junit-${filename}-report.xml || true
+  [[ -e "$tf" ]] || break  # handle the case of no *.wav files
+  filename=$(basename "${tf}" ${TARGET_FOLDER}/output.tap)
+  if [ -s "${tf}" ]; then
+    tap-junit --package="Agent Node.js" > "${TARGET_FOLDER}/junit-${filename}-report.xml" < "${tf}" || true
```
Comment (Qard) [inline .ci/scripts/convert_tap_to_junit.sh]: Should `--package` be something different?

**Comment by Qard** [comment]
Is this in review? There are no zube labels on it. 🤔

**Comment by Qard** [comment]
The failures in Jenkins match what we've been seeing for awhile now in Travis, and are expected. It is a known issue with Node.js 6.x and will not be fixed as it is already end-of-life. Support for Node.js 6 will be dropped in the next major release.
